test/src/kbasesearchengine/test/events/handler/WorkspaceEventHandlerTest.java

```diff
@@ -1472,4 +1473,123 @@ public void newVersionEventWithOverwriteExistingData() throws Exception {
         Assert.assertEquals(expectedEvent, updatedEvent);
     }
 
+    /*
+    Test that a NEW_VERSION object event with the isPublic flag set to false will be updated
+    with the isPublic flag to true if the current state of the associated workspace
+    has the public permission field set to "r".
+    This triggers the isPublic logic in WorkspaceEventHandler.updateObjectEvent. 
+    */
+    @Test
+    public void newVersionPublicFlagUpdatedFromWorkspaceCase1() throws Exception {
+        final CloneableWorkspaceClient clonecli = mock(CloneableWorkspaceClient.class);
+        final WorkspaceClient cloned = mock(WorkspaceClient.class);
+        final WorkspaceClient wscli = mock(WorkspaceClient.class);
+        when(clonecli.getClientClone()).thenReturn(cloned);
+        when(clonecli.getClient()).thenReturn(wscli);
+
+        StatusEvent event = StatusEvent.getBuilder(
+                StorageObjectType.fromNullableVersion("WS", "foo", 1),
+                Instant.ofEpochMilli(20000),
+                StatusEventType.NEW_VERSION)
+                .withNullableOverwriteExistingData(true)  // overwrite existing data
+                .withNullableisPublic(Boolean.FALSE)
+                .withNullableAccessGroupID(3)
+                .withNullableObjectID("1")
+                .withNullableVersion(1)
+                .build();
+
+        List<Tuple11<Long, String, String, String,
+                Long, String, Long, String, String, Long, Map<String, String>>> objList = new ArrayList<Tuple11<Long, String, String, String, Long, String, Long, String, String, Long, Map<String,String>>>();
+        objList.add(objTuple(1L, null, "sometype", "date", 1L, "copier",
+                3L, "wsname", "checksum", 44, Collections.emptyMap()));
+
+        when(wscli.administer(any()))
```

Comment:
this could be a little unclear later on? maybe add the method in the comments too?

Reply:
I'm not sure what you mean here.

Reply:
This test was copied from an earlier test. I'm not too familiar with mockito, but I did spend a few minutes reading up on it so that I could understand how this mechanism works. If you are referring to the comment above the mockito stuff explaining how it is supposed to work, I've removed that.